h2o-r/h2o-package/R/classes.R

```diff
@@ -688,7 +688,7 @@ setMethod("show", "H2ORegressionMetrics", function(object) {
   } else {
       cat("Mean Residual Deviance :  ", h2o.mean_residual_deviance(object), "\n", sep="")
   }
-  if(!is.null(object@algorithm) && object@algorithm %in% c("glm","gbm","drf","xgboost","generic") && exists("r2", where=object@metrics)) {
+  if(!is.null(object@algorithm) && object@algorithm %in% c("glm","gbm","drf","xgboost","generic","deeplearning") && exists("r2", where=object@metrics)) {
```

Comment:
This is a bugfix. R^2 has not been shown for DeepLearning models.

Reply:
might be on purpose though, calculation of R^2 in non-GLM models is not right (there is a jira issue for that)

Reply:
So...should I remove it for DL Generic model instead ? Only for the DL one.

Reply:
I resolved it. This small error led me to a series of problems. First, model attributes were filled in two passes - one pass was done manually and just mapped between two DTOs. This pass did not contain mapping for two fields: mean per-class error and description.

Second pass was automatic mapping using reflection I've wrote and was unnecessary. It only mapped those two fields mentioned above which were missing in the programmatic manual mapping and was also filling r-squared everywhere.

Therefore, I got rid of the mapping by reflection completely, as we want control of what goes into these V3 objects. And we had it. Only fixed the two fields mapping which were not filled manually, but this was masked by the second pass of the automatic mapper. 

Now it works and R^2 for regression is only sent for GLM.